### ISSUE TITLE
References using ObjectId ?

### ISSUE THREAD

**Issue by kedare** [rs/rest-layer-mongo#6]
Hello,

Looks like it is not possible to used ObjectId as references ? When trying to use them as simple references, it tries to find the reference with the string of the ObjectID (ex: "58555ef4314b56679216dc91") instead of the good way to use ObjectId (ObjectId("58555ef4314b56679216dc91")).

Here is the current query from the profiler :

```
{
    "op" : "query",
    "ns" : "test.users",
    "query" : {
        "find" : "users",
        "filter" : {
            "_id" : "58555ef4314b56679216dc91"
        }
}
```

Here is how it should be :
```
{
    "op" : "query",
    "ns" : "test.users",
    "query" : {
        "find" : "users",
        "filter" : {
            "_id" : ObjectId("58555ef4314b56679216dc91")
        }
    }
}
```

Tested using a simple references :

```
			"owner": {
				Required:   true,
				Filterable: true,
				Validator: &schema.Reference{
					Path: "users",
				},
			}
```

And the user model :
```
	users = schema.Schema{
		Description: "User account",
		Fields: schema.Fields{
			"id":      mongo.ObjectIDField,
			"created": schema.CreatedField,
			"updated": schema.UpdatedField,
			"name": {
				Required:   true,
				Filterable: true,
				Validator: &schema.String{
					MaxLen: 150,
				},
			},
			"email": {
				Required:   true,
				Filterable: true,
				Validator: &schema.String{
					MaxLen: 150,
				},
			},
			"securityData": {
				Required:   true,
				Filterable: true,
			},
		},
          }
	
```
Is there any way to fix that ?

Thanks

**Comment by rs** [comment]
I found two bugs, one in rest-layer and another in the mongo storage. Fixes are coming.

**Comment by rs** [comment]
The fix in rest-layer is: e5aeb528ec4401194e4e007a429059830564e104
